core/server/models/post.js

```diff
@@ -509,8 +510,8 @@ Post = ghostBookshelf.Model.extend({
         }
         // If the current column settings allow it...
         if (!options.columns || (options.columns && options.columns.indexOf('primary_tag') > -1)) {
-            // ... attach a computed property of primary_tag which is the first tag or null
-            attrs.primary_tag = attrs.tags && attrs.tags.length > 0 ? attrs.tags[0] : null;
+            // ... attach a computed property of primary_tag which is the first public tag or null
+            attrs.primary_tag = visibility.first(attrs.tags, ['public']);
```

Comment:
I think `visibility.first` is returning `undefined` when no tag matches which is resulting in no `primary_tag` property being added causing the tests to fail.

I'm not sure if `visibility.first` should be returning `null` or if we need a `|| null` conditional here.